tests/python/pytorch/dataloading2/test_csc_sampling_graph.py

```diff
@@ -0,0 +1,124 @@
+import unittest
+
+import backend as F
+
+import numpy as np
+import pytest
+import torch
+
+from dgl.dataloading2 import *
+
+
+def random_heterogeneous_graph_from_csc(num_nodes, num_ntypes, num_etypes):
+    # assume each node has 0 ~ 10 neighbors
+    csc_indptr = torch.randint(0, 10, (num_nodes,))
+    csc_indptr = torch.cumsum(csc_indptr, dim=0)
+    csc_indptr = torch.cat((torch.tensor([0]), csc_indptr), dim=0)
+    num_edges = csc_indptr[-1].item()
+    indices = torch.randint(0, num_nodes, (num_edges,))
+    ntypes = [f"n{i}" for i in range(num_ntypes)]
+    etypes = [f"e{i}" for i in range(num_etypes)]
+    # random get node type split point
+    node_type_offset = torch.sort(torch.randperm(num_nodes)[:num_ntypes])[0]
+    node_type_offset[0] = 0
+    type_per_edge = torch.randint(0, num_etypes, (num_edges,))
+    return (
+        from_csc(
+            csc_indptr,
+            indices,
+            etype_sorted=False,
+            hetero_info=(ntypes, etypes, node_type_offset, type_per_edge),
+        ),
+        csc_indptr,
+        indices,
+        type_per_edge,
+    )
+
+
+def random_heterogeneous_graph_from_coo(
+    num_nodes, num_edges, num_ntypes, num_etypes
+):
+    coo = torch.randint(0, num_nodes, (2, num_edges))
+    ntypes = [f"n{i}" for i in range(num_ntypes)]
+    etypes = [f"e{i}" for i in range(num_etypes)]
+    type_per_edge = torch.randint(0, num_etypes, (num_edges,))
+    node_type_offset = torch.sort(torch.randperm(num_nodes)[:num_ntypes])[0]
+    node_type_offset[0] = 0
+    return (
+        from_coo(
+            coo,
+            hetero_info=(ntypes, etypes, node_type_offset, type_per_edge),
+        ),
+        torch.cat([coo, type_per_edge.unsqueeze(dim=0)], dim=0),
+    )
+
+
+def sort_coo(coo: torch.tensor):
+    row, col, etype = coo
+    indices = torch.from_numpy(
+        np.lexsort((etype.numpy(), row.numpy(), col.numpy()))
+    )
+    return torch.index_select(coo, 1, indices)
+
+
+@unittest.skipIf(
+    F._default_context_str == "gpu",
+    reason="Graph is CPU only at present.",
+)
+@pytest.mark.parametrize("num_nodes", [10, 50, 1000, 10000])
+def test_from_csc(num_nodes, num_ntypes=3, num_etypes=5):
```

Comment:
Can you test more corner cases?

* num_ntypes=num_etypes=0 (homogenous)
* num_ntypes=1, num_etypes=many (KG)
* edge types sharing the same name, e.g., ('user', 'like', 'movie'), ('user', 'like', book'), etc.
* types without any nodes or edges
* ...

You should think hard on making your code robust.